airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/ReplicationActivityImpl.java

```diff
@@ -147,33 +171,90 @@ private static StandardSyncOutput reduceReplicationOutput(final ReplicationOutpu
-          processFactory,
-          syncInput.getResourceRequirements());
-      final IntegrationLauncher destinationLauncher = new AirbyteIntegrationLauncher(
-          destinationLauncherConfig.getJobId(),
-          Math.toIntExact(destinationLauncherConfig.getAttemptId()),
-          destinationLauncherConfig.getDockerImage(),
-          processFactory,
-          syncInput.getResourceRequirements());
-
-      // reset jobs use an empty source to induce resetting all data in destination.
-      final AirbyteSource airbyteSource =
-          sourceLauncherConfig.getDockerImage().equals(WorkerConstants.RESET_JOB_SOURCE_DOCKER_IMAGE_STUB) ? new EmptyAirbyteSource()
-              : new DefaultAirbyteSource(sourceLauncher);
-
-      return new DefaultReplicationWorker(
-          jobRunConfig.getJobId(),
-          Math.toIntExact(jobRunConfig.getAttemptId()),
-          airbyteSource,
-          new NamespacingMapper(syncInput.getNamespaceDefinition(), syncInput.getNamespaceFormat(), syncInput.getPrefix()),
-          new DefaultAirbyteDestination(destinationLauncher),
-          new AirbyteMessageTracker(),
-          new AirbyteMessageTracker());
+    return () -> new Worker<>() {
+
+      final AtomicBoolean cancelled = new AtomicBoolean(false);
+      Process process;
+
+      @Override
+      public ReplicationOutput run(StandardSyncInput standardSyncInput, Path jobRoot) throws WorkerException {
+        try {
+          final Path jobPath = WorkerUtils.getJobRoot(workspaceRoot, jobRunConfig.getJobId(), jobRunConfig.getAttemptId());
+
+          final Map<String, String> fileMap = Map.of(
+              INIT_FILE_APPLICATION, REPLICATION,
+              INIT_FILE_JOB_RUN_CONFIG, Jsons.serialize(jobRunConfig),
+              INIT_FILE_SOURCE_LAUNCHER_CONFIG, Jsons.serialize(sourceLauncherConfig),
+              INIT_FILE_DESTINATION_LAUNCHER_CONFIG, Jsons.serialize(destinationLauncherConfig),
+              INIT_FILE_SYNC_INPUT, Jsons.serialize(syncInput),
+              INIT_FILE_ENV_MAP, Jsons.serialize(System.getenv()));
+
+          // for now keep same failure behavior where this is heartbeating and depends on the parent worker to
+          // exist
+          process = processFactory.create(
+              "runner-" + UUID.randomUUID().toString().substring(0, 10),
+              0,
+              jobPath,
+              "airbyte/runner:" + airbyteVersion,
+              false,
+              fileMap,
+              null,
+              workerConfigs.getResourceRequirements(),
+              Map.of(KubeProcessFactory.JOB_TYPE, KubeProcessFactory.SYNC_RUNNER),
+              Map.of(WorkerApp.KUBE_HEARTBEAT_PORT, WorkerApp.KUBE_HEARTBEAT_PORT));
+
+          final AtomicReference<ReplicationOutput> output = new AtomicReference<>();
+
+          LineGobbler.gobble(process.getInputStream(), line -> {
+            final var maybeOutput = Jsons.tryDeserialize(line, ReplicationOutput.class);
```

Comment:
Not something that I want to enforce because doing the review with IntelliJ fixes the issue but I would like to avoid using var (or val) when the type is not explicit (like a return in this case). It is a small detail but it can make the review process easier.